blockbuster/lane_abci.go

```diff
@@ -0,0 +1,67 @@
+package blockbuster
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/skip-mev/pob/blockbuster/utils"
+)
+
+// PrepareLane will prepare a partial proposal for the lane. It will select transactions from the
+// lane respecting the selection logic of the prepareLaneHandler. It will then update the partial
+// proposal with the selected transactions. If the proposal is unable to be updated, we return an
+// error. The proposal will only be modified if it passes all of the invarient checks.
+func (l *LaneConstructor) PrepareLane(
+	ctx sdk.Context,
+	proposal BlockProposal,
+	maxTxBytes int64,
+	next PrepareLanesHandler,
+) (BlockProposal, error) {
+	txs, txsToRemove, err := l.prepareLaneHandler(ctx, proposal, maxTxBytes)
+	if err != nil {
+		return proposal, err
+	}
+
+	// Remove all transactions that were invalid during the creation of the partial proposal.
+	if err := utils.RemoveTxsFromLane(txsToRemove, l); err != nil {
+		l.Logger().Error(
+			"failed to remove transactions from lane",
+			"lane", l.Name(),
+			"err", err,
+		)
+	}
+
+	// Update the proposal with the selected transactions.
+	if err := proposal.UpdateProposal(l, txs); err != nil {
```

Comment:
Feel like the PrepareLane methods shld only return txs (and the top level Chainer would handle constructing the proposal). I don't like how the proposal / insertion logic is passed into each lane's PrepareLane

Reply:
resolved offline